lib/render/SliceRenderer.cpp

```diff
@@ -176,284 +164,60 @@ void SliceRenderer::_resetColormapCache()
-        if (y >= _cacheParams.boxMin[Y] && y <= _cacheParams.boxMax[Y]) {
-            _vertexIn2dAnd3d p = {glm::vec3(x, y, z), glm::vec2()};
-            vertices.push_back(p);
-        }
-    };
-    auto xIntercept = [&](float y, float z) {
-        if (_normal.x == 0) return;
-        double x = (_normal.x * _origin.x + _normal.y * _origin.y + _normal.z * _origin.z - _normal.y * y - _normal.z * z) / _normal.x;
-        if (x >= _cacheParams.boxMin[X] && x <= _cacheParams.boxMax[X]) {
-            _vertexIn2dAnd3d p = {glm::vec3(x, y, z), glm::vec2()};
-            vertices.push_back(p);
-        }
-    };
-
-    // Find vertices that exist on the Z edges of the Box
-    zIntercept(_cacheParams.boxMin[X], _cacheParams.boxMin[Y]);
-    zIntercept(_cacheParams.boxMax[X], _cacheParams.boxMax[Y]);
-    zIntercept(_cacheParams.boxMin[X], _cacheParams.boxMax[Y]);
-    zIntercept(_cacheParams.boxMax[X], _cacheParams.boxMin[Y]);
-    // Find any vertices that exist on the Y edges of the Box
-    yIntercept(_cacheParams.boxMin[X], _cacheParams.boxMin[Z]);
-    yIntercept(_cacheParams.boxMax[X], _cacheParams.boxMax[Z]);
-    yIntercept(_cacheParams.boxMin[X], _cacheParams.boxMax[Z]);
-    yIntercept(_cacheParams.boxMax[X], _cacheParams.boxMin[Z]);
-    // Find any vertices that exist on the X edges of the Box
-    xIntercept(_cacheParams.boxMin[Y], _cacheParams.boxMin[Z]);
-    xIntercept(_cacheParams.boxMax[Y], _cacheParams.boxMax[Z]);
-    xIntercept(_cacheParams.boxMin[Y], _cacheParams.boxMax[Z]);
-    xIntercept(_cacheParams.boxMax[Y], _cacheParams.boxMin[Z]);
-}
-
-stack<glm::vec2> SliceRenderer::_2DConvexHull(std::vector<_vertexIn2dAnd3d> &vertices) const
+int SliceRenderer::_regenerateSlice()
 {
-    VAssert(vertices.size() > 0);
-
-    // We now have a set of vertices along the Box's XYZ intercepts.  The edges of these vertices define where
-    // the user should see data.  To find the connectivity/edges of these vertices, we will first project them into a 2D
-    // coordinate system using our basis function, and then perform Convex Hull on those 2D coordinates.
-
-    // Project our 3D points onto the 2D plane using our basis function (_normal, _axis1, and _axis2)
-    glm::vec2 *unorderedTwoDPoints = new glm::vec2[vertices.size()];
-    int        count = 0;
-    for (auto &vertex : vertices) {
-        double x = glm::dot(_axis1, vertex.threeD - _origin);    // Find 3D point's projected X coordinate
-        double y = glm::dot(_axis2, vertex.threeD - _origin);    // Find 3D point's projected Y coordinate
-        vertex.twoD = {x, y};
-        unorderedTwoDPoints[count].x = vertex.twoD.x;
-        unorderedTwoDPoints[count].y = vertex.twoD.y;
-        count++;
+    Grid* grid3d = nullptr;
+    int rc = _getGrid3D( grid3d );
+    if ( rc < 0 ) return -1;
+
+    // Get data values from a slice
+    float* dataValues = new float[_textureSideSize*_textureSideSize];
```

Comment:
It doesn't look like this memory is ever being freed. is there a reason not to use a smart pointer?

Reply:
An oversight.  The function prototype I was given in the last review was using a raw pointer and using a smart ptr didn't occur to me.  Fixed.